### PR TITLE
[C++] Competitive (LeetCode & SPOJ)

### PR DESCRIPTION
**Pull request by trivediavinash**
I have added the new solutions in C++/competitive

### CODE REVIEW

**Comment by trivediavinash** [comment]
@amanmehara  Please review my PR

File: C++/competitive/LeetCode/maximum-subarray.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
+        // Copyright 2020  Avinash  Trivedi
```
Comment (amanmehara) [inline C++/competitive/LeetCode/maximum-subarray.cpp:1]: Align the licence header correctly.

File: C++/competitive/SPOJ/CSUMQ-Cumulative-Sum-Query.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
+        // Copyright 2020  Avinash  Trivedi
```
Comment (amanmehara) [inline C++/competitive/SPOJ/CSUMQ-Cumulative-Sum-Query.cpp:1]: Align the licence header correctly.

File: C++/competitive/SPOJ/CSUMQ-Cumulative-Sum-Query.cpp [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+ 			for (intt i = x; i <= y; i++)
+ 			{
+ 				sum += a[i];
+ 			}
+
+ 			cout << sum << endl;
+ 		}
+
+ 		return 0;
+ 	}
```
Comment (amanmehara) [inline C++/competitive/SPOJ/CSUMQ-Cumulative-Sum-Query.cpp:60]: Add newline at end of file.

File: C++/competitive/LeetCode/Best-Time-to-Buy-and-Sell-Stock-II.cpp [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+					sum += prices[i + 1] - prices[i];
+
+				}
+			}
+
+			//Finally return the sum.
+			return sum;
+
+		}
+};
```
Comment (amanmehara) [inline C++/competitive/LeetCode/Best-Time-to-Buy-and-Sell-Stock-II.cpp:43]: Add newline at end of file.

File: C++/competitive/LeetCode/maximum-subarray.cpp [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ 				ms = max(cs, ms);
+ 				if (cs < 0)
+ 				{
+ 					cs = 0;
+ 				}
+ 			}
+
+ 			return ms;
+ 		}
+ };
```
Comment (amanmehara) [inline C++/competitive/LeetCode/maximum-subarray.cpp:42]: Add newline at end of file.

**Comment by amanmehara** [comment]
Please have a look at https://github.com/amanmehara/programming/blob/master/CONTRIBUTING.md and make changes accordingly.
I'll approve and merge afterwards! 😃

File: C++/competitive/LeetCode/maximum-subarray.cpp (hunk 5 above)
Comment (trivediavinash) [reply C++/competitive/LeetCode/maximum-subarray.cpp:42]: @amanmehara should i add extra line after line no 42

Comment (amanmehara) [reply C++/competitive/LeetCode/maximum-subarray.cpp:42]: Yes

**Comment by trivediavinash** [comment]
@amanmehara  I have done the changes .

File: C++/competitive/LeetCode/maximum-subarray.cpp [hunk 6]
```diff
@@ -0,0 +1,40 @@
+// Copyright 2020  Avinash  Trived
```
Comment (amanmehara) [inline C++/competitive/LeetCode/maximum-subarray.cpp:1]: Your name is incorrect! 😛

File: C++/competitive/SPOJ/CSUMQ-Cumulative-Sum-Query.cpp [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+ 			intt sum = 0;
+ 			for (intt i = x; i <= y; i++)
+ 			{
+ 				sum += a[i];
+ 			}
+
+ 			cout << sum << endl;
+ 		}
+
+ 		return 0;
```
Comment (amanmehara) [inline C++/competitive/SPOJ/CSUMQ-Cumulative-Sum-Query.cpp:56]: Add some comments explaining the steps.

**Review by amanmehara — CHANGES_REQUESTED** [review]
@trivediavinash The changes are good. 😃 
Some minor comments.
I'll approve and merge after they are fixed.

File: C++/competitive/SPOJ/CSUMQ-Cumulative-Sum-Query.cpp (hunk 7 above)
Comment (trivediavinash) [reply C++/competitive/SPOJ/CSUMQ-Cumulative-Sum-Query.cpp]: @amanmehara Done with the changes

**Pull request merged**